applications/portal/backend/api/management/utilities.py

```diff
@@ -1,6 +1,22 @@
-from api.models import AtlasesChoice
+from django.db import transaction
+
+from api.models import AtlasesChoice, Population, PopulationStatus, Experiment
 
 
 def get_valid_atlases():
     valid_atlas_ids = {e.value for e in AtlasesChoice}
     return valid_atlas_ids
+
+
+def register_experiment(data_filepath, experiment_id, population_names):
+    populations = []
+    with transaction.atomic():
+        for name in population_names:
+            populations.append(Population.objects.create(
+                experiment_id=experiment_id,
+                name=name,
+                is_fiducial=False,
+                status=PopulationStatus.PENDING
+            ))
+    experiment = Experiment.objects.get(pk=experiment_id)
```

Comment:
question: @afonsobspinto why didn't you wrap the register not in the transaction?

Reply:
The register can take hours to finish and I wanted to avoid having a 'transaction' open for so long.

Reply:
Also, because Population has a state, I think we can make the case that the creation of the populations is 'independent' of the registration being successful or not

Reply:
> The register can take hours to finish and I wanted to avoid having a 'transaction' open for so long.

okay